modin/core/io/text/csv_glob_dispatcher.py

```diff
@@ -348,14 +345,27 @@ def file_exists(cls, file_path: str, storage_options=None) -> bool:
                 else:
                     new_storage_options = {}
 
-                s3fs = S3FS.S3FileSystem(anon=False, **new_storage_options)
+                fs, _ = fsspec.core.url_to_fs(file_path, **new_storage_options)
                 exists = False
                 try:
-                    exists = len(s3fs.glob(file_path)) > 0 or exists
-                except (NoCredentialsError, PermissionError, EndpointConnectionError):
+                    exists = fs.exists(file_path)
+                except (
+                    NoCredentialsError,
+                    PermissionError,
+                    EndpointConnectionError,
+                    ConnectTimeoutError,
+                ):
                     pass
-                s3fs = S3FS.S3FileSystem(anon=True, **new_storage_options)
-                return exists or len(s3fs.glob(file_path)) > 0
+                fs, _ = fsspec.core.url_to_fs(
+                    file_path, anon=True, **new_storage_options
+                )
```

Comment:
let's do anonymous check in `except` block only; also it might make sense to see if `anon` wasn't already in `new_storage_options`

Reply:
Moved to `except` block. `new_storage_options` doesn't have `anon` because we have specific check above to handling this